libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -726,6 +726,13 @@ bool AP_InertialSensor::register_gyro(uint8_t &instance, uint16_t raw_sample_rat
         return false;
     }
 
+    // if we have already registered this sensor then don't register it again
+    for (uint8_t i=0; i<_gyro_count; i++) {
+        if ((uint32_t)_gyro_id(i) == id) {
```

Comment:
this will stop it registering, but the driver will still be there ie. the backend will still be there in the list, _backend_count will still include it, so update() will still be called on the backend

Reply:
@tridge I have changed the strategy for detecting remaining extra sensors now.